sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs

```diff
@@ -210,6 +221,7 @@ public BlobContainerClient(string connectionString, string blobContainerName, Bl
             _clientDiagnostics = new ClientDiagnostics(options);
             _customerProvidedKey = options.CustomerProvidedKey;
             _encryptionScope = options.EncryptionScope;
+            _storageSharedKeyCredential = StorageSharedKeyCredential.ParseConnectionString(connectionString);
```

Comment:
Ditto on reusing `conn.Credentials as StorageSharedKeyCredential` across the board.